app/services/expired_registration_cleanup_service.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+class ExpiredRegistrationCleanupService < ::WasteExemptionsEngine::BaseService
+  def run
+    return unless registration_ids.any?
+
+    ActiveRecord::Base.transaction do
+      # paper_trail will try creating a record for the deletion
+      # unless we execute within this block
+      PaperTrail.request(enabled: false) do
+        clear_registrations!
+      end
+      clear_versions!
+      clear_versions_archive!
+    end
+  end
+
+  private
+
+  # registration exemptions that were deregistered (ceased or revoked)
+  # or expired over 7 years ago
+  def registration_ids
+    @registration_ids ||=
+      WasteExemptionsEngine::RegistrationExemption
+      .where(
+        <<~SQL
+          deregistered_at <= '#{date}'
+          OR (state = 'expired' AND updated_at <= '#{date}')
+        SQL
+      )
+      .pluck(:registration_id)
+      .uniq
+  end
+
+  def date
+    @date ||= 7.years.ago.to_date
+  end
+
+  def clear_registrations!
```

Comment:
Can you log the registration number during deleting please? this will allow me to check specifically what registrations have been deleted.

Reply:
@timstone just checking, do you mean the regIdentifier as opposed to the database ID?

Reply:
Yes the regIdentifier, sorry if that wasn't clear.

Reply:
references go to `Rails.logger.info` now 

https://github.com/DEFRA/waste-exemptions-back-office/pull/845/files#diff-3413aedef72933aa883008556057ff4cf68ffbda1ef3d8dd775ff1fd3ba0d548R39-R48